### PR TITLE
vochain: add farcaster frame support

### PR DESCRIPTION
**Pull request by p4u**
Incorporate a verifier for farcaster frame based voting.

TODO: ensure the processID of the signed frame message corresponds the processID. Right now there is no way to incorporate the processID information into the frame signed message body (maybe the URL?).

### CODE REVIEW

File: vochain/transaction/election_tx.go [hunk 1]
```diff
@@ -43,9 +47,23 @@ func (t *TransactionHandler) NewProcessTxCheck(vtx *vochaintx.Tx) (*models.Proce
...
+	case models.CensusOrigin_OFF_CHAIN_CA:
+		if tx.Process.EnvelopeType.Anonymous {
+			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not supported for CSP voting")
+		}
+	case models.CensusOrigin_FARCASTER_FRAME:
+		if tx.Process.EnvelopeType.Anonymous {
+			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not supported for Farcaster voting")
+		}
+		if tx.Process.EnvelopeType.EncryptedVotes {
+			return nil, ethereum.Address{}, fmt.Errorf("encrypted votes not supported for Farcaster voting")
```
Comment (jordipainan) [inline vochain/transaction/election_tx.go:62]:
```suggestion
			return nil, ethereum.Address{}, fmt.Errorf("encrypted votes not yet supported for Farcaster voting")
```

File: vochain/transaction/election_tx.go [hunk 2]
```diff
@@ -43,9 +47,23 @@ func (t *TransactionHandler) NewProcessTxCheck(vtx *vochaintx.Tx) (*models.Proce
...
+
+	// run specific checks based on census origin
+	switch tx.Process.CensusOrigin {
+	case models.CensusOrigin_OFF_CHAIN_CA:
+		if tx.Process.EnvelopeType.Anonymous {
+			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not supported for CSP voting")
+		}
+	case models.CensusOrigin_FARCASTER_FRAME:
+		if tx.Process.EnvelopeType.Anonymous {
+			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not supported for Farcaster voting")
```
Comment (jordipainan) [inline vochain/transaction/election_tx.go:59]:
```suggestion
			return nil, ethereum.Address{}, fmt.Errorf("anonymous process not yet supported for Farcaster voting")
```

File: vochain/transaction/election_tx.go (hunk 1 above)
Comment (p4u) [reply vochain/transaction/election_tx.go:62]: Sorry to say, we cannot support secret until the end. The blockchain only receives buttonIndex=N as part of the proof and we need to verify buttonIndex == votePackage. So votePacakge cannot be encrypted.

File: vochain/transaction/proofs/farcasterproof/farcasterproof.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	if msg.Data == nil {
+		return nil, nil, fmt.Errorf("invalid message data")
+	}
+	if msg.SignatureScheme != farcasterpb.SignatureScheme_SIGNATURE_SCHEME_ED25519 {
+		return nil, nil, fmt.Errorf("invalid signature scheme")
+	}
+	if msg.Data.Type != farcasterpb.MessageType_MESSAGE_TYPE_FRAME_ACTION {
+		return nil, nil, fmt.Errorf("invalid message type, got %s", msg.Data.Type.String())
+	}
+	var pubkey ed25519.PublicKey = msg.GetSigner()
```
Comment (jordipainan) [inline vochain/transaction/proofs/farcasterproof/farcasterproof.go]:
```suggestion
	pubkey := msg.GetSigner()
```

**Review by jordipainan — COMMENTED** [review]
lgtm, just some nits !

**Pull request merged**